sympy/logic/boolalg.py

```diff
@@ -1136,6 +1148,14 @@ def _eval_simplify(self, **kwargs):
         return self._apply_patternbased_simplification(rv, patterns,
             kwargs['measure'], None)
 
+    def _eval_subs(self, old, new):
+        # If old is Xor, replace the parts of the arguments with new if all
+        # are there
+        if isinstance(old, Xor):
+            if all(a in self.args for a in old.args):
+                args = [new] + [a for a in self.args  if a not in old.args]
+                return self.func(*args)
+
```

Comment:
Might the following be better? Not sure:
```python
>>> o={1,2}
>>> s={1,2,3}
>>> sd = o.symmetric_difference(s)
>>> if not sd & old:  # all args in s are in o
...  print(sd)  # these are the args in s that are not in o
...
{3}
```

Reply:
Probably! Felt a bit wrong to write it like that, but too tired to really think of anything else than brute force...

Reply:
I find the logic with symmetric difference confusing. Basically this is:
```python
if old_args.issubset(self_args):
    self_args -= old_args
    self_args.add(new)
```

Reply:
Note that `issubset` can be called with a list/tuple as argument:
```python
In [38]: {1}.issubset((1, 2, 3))
Out[38]: True

In [39]: {1}.issubset((2, 2, 3))
Out[39]: False
```
That means that the common case (condition is false) can be a fast-path that doesn't create any new sets apart from `set(old.args)`.

Reply:
I find both set notations more confusing than the original list comprehension, but I can of course change to any of them.

Reply:
I have been confused by `symmetric_difference` but now think of it as the non-trivial difference between two sets: it gives the difference no matter what order you put the args, kind of like taking `abs(x-y)` for two scalars. 

I like @oscarbenjamin suggestion:
```
old_set = set(old)
if old_set.is_subset(args):
    args = set(args) - old_set
    args.update(new)
    return self.func(*args)
```

(with the last line unindented in Xor)